spec/controllers/trainer_controller_spec.rb

```diff
@@ -1,268 +1,85 @@
-        card = check_review_card(card, 'RoR', 1)
-        card = check_review_card(card, 'house', 1)
-        expect(card.efactor).to eq(2.28)
-        expect(card.quality).to eq(4)
-      end
-
-      it 'repeat=2 quality=3' do
-        card = create(:card, user: @user, block: @block,
-                      interval: 6, repeat: 2, efactor: 2.6, quality: 5)
-        card = check_review_card(card, 'RoR', 2)
-        card = check_review_card(card, 'house', 1)
-        expect(card.efactor).to eq(1.6)
-        expect(card.quality).to eq(3)
-      end
-
-      it 'repeat=2 quality=3' do
-        card = create(:card, user: @user, block: @block,
-                      interval: 6, repeat: 2, efactor: 2.6, quality: 5)
-        card = check_review_card(card, 'RoR', 3)
-        card = check_review_card(card, 'house', 1)
-        expect(card.efactor).to eq(1.3)
-        expect(card.quality).to eq(3)
-      end
-
-      it 'repeat=3 quality=5' do
-        card = create(:card, user: @user, block: @block,
-                      interval: 16, repeat: 3, efactor: 2.7, quality: 5)
-        card = check_review_card(card, 'house', 1)
-        expect(card.review_date.strftime('%Y-%m-%d %H:%M')).
-            to eq((Time.zone.now + 16.days).strftime('%Y-%m-%d %H:%M'))
-        expect(card.interval).to eq(45)
-        expect(card.repeat).to eq(4)
-        expect(card.attempt).to eq(1)
-      end
-
-      it 'repeat=3 quality=5' do
-        card = create(:card, user: @user, block: @block,
-                      interval: 16, repeat: 3, efactor: 2.7, quality: 5)
-        card = check_review_card(card, 'house', 1)
-        expect(card.efactor).to eq(2.8)
-        expect(card.quality).to eq(5)
-      end
-
-      it 'repeat=3 quality=4' do
-        card = create(:card, user: @user, block: @block,
-                      interval: 16, repeat: 3, efactor: 2.7, quality: 5)
-        card = check_review_card(card, 'RoR', 1)
-        card = check_review_card(card, 'house', 1)
-        expect(card.efactor).to eq(2.38)
-        expect(card.quality).to eq(4)
-      end
-
-      it 'repeat=3 quality=3' do
-        card = create(:card, user: @user, block: @block,
-                      interval: 16, repeat: 3, efactor: 2.7, quality: 5)
-        card = check_review_card(card, 'RoR', 2)
-        card = check_review_card(card, 'house', 1)
-        expect(card.efactor).to eq(1.7)
-        expect(card.quality).to eq(3)
-      end
-
-      it 'repeat=3 quality=3' do
-        card = create(:card, user: @user, block: @block,
-                      interval: 16, repeat: 3, efactor: 2.7, quality: 5)
-        card = check_review_card(card, 'RoR', 3)
-        card = check_review_card(card, 'house', 1)
-        expect(card.efactor).to eq(1.3)
-        expect(card.quality).to eq(3)
-      end
+  let(:user)  { create(:user) }
+  let(:register) { @controller.send(:auto_login, user) }
+  let(:block) { create(:block, user: user) }
+  let(:card)  { create(:card, user: user,
```

Comment:
Avoid using {...} for multi-line blocks.<br>Unnecessary spacing detected.<br>Block body expression is on the same line as the block start.